cgnsutilities/cgnsutilities.py

```diff
@@ -1631,34 +1663,46 @@ def coarsen(self):
         conditions and B2B if necessary"""
         # We will coarsen one direction at a time. We do this to check if the block
         # is already 1-cell wide, which can't be coarsened any further
-        if self.dims[0] != 2:
-            self.coords = self.coords[0::2, :, :, :]
-            # Update BCs and B2B
-            for boco in self.bocos:
-                boco.coarsen(0)
-            for b2b in self.B2Bs:
-                b2b.coarsen(0)
-
-        if self.dims[1] != 2:
-            self.coords = self.coords[:, 0::2, :, :]
-            # Update BCs and B2B
-            for boco in self.bocos:
-                boco.coarsen(1)
-            for b2b in self.B2Bs:
-                b2b.coarsen(1)
-
-        if self.dims[2] != 2:
-            self.coords = self.coords[:, :, 0::2, :]
-            # Update BCs and B2B
-            for boco in self.bocos:
-                boco.coarsen(2)
-            for b2b in self.B2Bs:
-                b2b.coarsen(2)
-
-        # Update dimensions
-        self.dims[0] = self.coords.shape[0]
-        self.dims[1] = self.coords.shape[1]
-        self.dims[2] = self.coords.shape[2]
+
+        # the new dimensions are half rounded up of the old dimensions
+        new_dims = copy.deepcopy(self.dims)
+        for i in range(3):
+            if self.dims[i] > 2:
+                new_dims[i] = (self.dims[i] + 1) // 2
+        new_coords = numpy.zeros((new_dims[0], new_dims[1], new_dims[2], 3))
```

Comment:
Should we add some type of warning or message that is printed to let the user know if a particular dimension could not be "perfectly" coarsened? The intent seems that when coarsening for instance 1x4 `|-|-|-|` would be coarsened to 1x2 `|-|--|` with a larger second cell. See the attached figure.

Reply:
A message has been added when a dimension cannot be evenly coarsened.

The shortening issue when coarsening from 4 or 3 to 2 has now been fixed too.